src/test/java/com/speaker/service/MessageServiceImplTest.java

```diff
@@ -0,0 +1,131 @@
+    @Mock
+    private MessageConvertor messageConvertor;
+    @InjectMocks
+    private MessageServiceImpl messageService;
+
+    @Test
+    void getMessagesByAccountIdSuccess() {
+        List<Message> messages = generateMessageList();
+        when(messageRepository.findAllByAccountId(ACCOUNT_ID)).thenReturn(messages);
+        assertThat(messageService.getMessagesByAccountId(ACCOUNT_ID), is(messages));
+    }
+
+    @Test
+    void addingMessageSuccess() {
+        List<Message> message = generateMessageList();
+        List<MessageDTO> messageDTO = generateMessageDTOList(ACCOUNT_FIRST_LAST_NAME, FRIEND_FIRST_LAST_NAME);
+        when(accountRepository.findAccountIdByNameAndLastName(ACCOUNT_FIRST_NAME, LAST_NAME)).thenReturn(Optional.of(ACCOUNT_ID));
+        when(accountRepository.findAccountIdByNameAndLastName(FRIEND_FIRST_NAME, LAST_NAME)).thenReturn(Optional.of(FRIEND_ID));
+        when(messageConvertor.convertToMessage(messageDTO.get(INDEX_0), FRIEND_ID, ACCOUNT_ID)).thenReturn(message.get(INDEX_0));
+        when(messageConvertor.convertToMessage(messageDTO.get(1), FRIEND_ID, ACCOUNT_ID)).thenReturn(message.get(INDEX_1));
+        when(messageRepository.createMessages(message)).thenReturn(INSERT_TRUE);
+        assertThat(messageService.addMessage(messageDTO), is(Response.TRUE));
+    }
+
+    @Test
+    void messageNotAddingWhenAccountIdIsNull() {
+        List<MessageDTO> messageDTO = generateMessageDTOList(ACCOUNT_FIRST_LAST_NAME, FRIEND_FIRST_LAST_NAME);
+        List<Message> message = generateMessageList();
+        when(accountRepository.findAccountIdByNameAndLastName(ACCOUNT_FIRST_NAME, LAST_NAME)).thenReturn(Optional.empty());
+        when(accountRepository.findAccountIdByNameAndLastName(FRIEND_FIRST_NAME, LAST_NAME)).thenReturn(Optional.of(FRIEND_ID));
+        assertThat(messageService.addMessage(messageDTO), is(Response.FALSE));
+        verify(accountRepository, times(2)).findAccountIdByNameAndLastName(ACCOUNT_FIRST_NAME, LAST_NAME);
+        verify(accountRepository, times(2)).findAccountIdByNameAndLastName(FRIEND_FIRST_NAME, LAST_NAME);
+        verify(messageConvertor, never()).convertToMessage(messageDTO.get(INDEX_0), null, FRIEND_ID);
+        verify(messageConvertor, never()).convertToMessage(messageDTO.get(INDEX_1), null, FRIEND_ID);
+        verify(messageRepository, never()).createMessages(message);
+    }
+
+    @Test
+    void messageNotAddingWhenAccountNameIsNull() {
+        List<MessageDTO> messageDTO = generateMessageDTOList(null, FRIEND_FIRST_LAST_NAME);
+        List<Message> message = generateMessageList();
+        assertThat(messageService.addMessage(messageDTO), is(Response.FALSE));
+        verify(accountRepository, never()).findAccountIdByNameAndLastName(null, null);
+        verify(accountRepository, times(2)).findAccountIdByNameAndLastName(FRIEND_FIRST_NAME, LAST_NAME);
+        verify(messageConvertor, never()).convertToMessage(messageDTO.get(INDEX_0), null, FRIEND_ID);
```

Comment:
the same

Reply:
Comments accepted, corrections made.  "Verify" has been deleted